From 27929860139c1ac29bf0c862c0fc416ea7020ae7 Mon Sep 17 00:00:00 2001
From: Rong Bao <webmaster@csmantle.top>
Date: Thu, 30 Apr 2026 17:34:41 +0800
Subject: [PATCH 16/19] Bug 2023167 - Part 8: Increase stack snapshot size on
 {loongarch64,riscv64}-linux. r=#profiler-reviewers!

The stack size needed on these platforms are much larger than that on
x64, so allocating more space for it.

Differential Revision: https://phabricator.services.mozilla.com/D297846
---
 tools/profiler/lul/LulMain.h | 18 ++++++++++++------
 1 file changed, 12 insertions(+), 6 deletions(-)

diff --git a/tools/profiler/lul/LulMain.h b/tools/profiler/lul/LulMain.h
index b634d9eec8308d53560483dfea54a3ba5f813b4b..f29e0f1ee533672ef7c6e567656af1e44c186e1b 100644
--- a/tools/profiler/lul/LulMain.h
+++ b/tools/profiler/lul/LulMain.h
@@ -167,13 +167,19 @@ struct UnwindRegs {
 #endif
 };
 
-// The maximum number of bytes in a stack snapshot.  This value can be increased
-// if necessary, but testing showed that 160k is enough to obtain good
-// backtraces on x86_64 Linux.  Most backtraces fit comfortably into 4-8k of
-// stack space, but we do have some very deep stacks occasionally.  Please see
-// the comments in DoNativeBacktrace as to why it's OK to have this value be so
-// large.
+// The maximum number of bytes in a stack snapshot.
+#if defined(GP_PLAT_loongarch64_linux) || defined(GP_PLAT_riscv64_linux)
+// On loongarch64-linux and riscv64-linux, stacks are generally larger, perhaps
+// due to there being more registers to save and the alignment requirement.
+static const size_t N_STACK_BYTES = 256 * 1024;
+#else
+// This value can be increased if necessary, but testing showed that 160k is
+// enough to obtain good backtraces on x86_64 Linux.  Most backtraces fit
+// comfortably into 4-8k of stack space, but we do have some very deep stacks
+// occasionally.  Please see the comments in DoNativeBacktrace as to why it's
+// OK to have this value be so large.
 static const size_t N_STACK_BYTES = 160 * 1024;
+#endif  // defined(GP_PLAT_loongarch64_linux) || defined(GP_PLAT_riscv64_linux)
 
 // The stack chunk image that will be unwound.
 struct StackImage {
-- 
2.55.0

